### ISSUE TITLE
WebResponse does not override headers & cookies when merging

### ISSUE THREAD

**Issue by thomasbachem** [agavi/agavi#1567]
I just stumbled upon a pretty unexpected issue: `WebResponse::merge()` doesn't merge headers and cookies that are already set on the target response (see https://github.com/agavi/agavi/blob/master/src/response/AgaviWebResponse.class.php#L305).

Is there a good reason for this?

**Comment by dzuelke** [comment]
Yes; the reason is that responses from slots get merged into the parent, but should not overwrite headers or cookies in the parent if they've been set there already.

**Comment by thomasbachem** [comment]
Hm, my use case is this: There's a default value for a header that is set in the parent, but the children can override it. Is there another way to set a default value for a header otherwise?
